8.cytopick_analysis/scripts/prediction_images_find.py

```diff
@@ -0,0 +1,532 @@
+radius = 50
+# define the number of cells to select
+n = 5
+
+
+# In[12]:
+
+
+dict_of_subset_dfs = {}
+for key in tqdm(dict_of_dfs):
+    df = dict_of_dfs[key]
+    if len(df) == 0:
+        pass
+    else:
+        # select n random cells from the dataframe
+        df = random_cell_select(df, n)
+        # add the df to the dictionary
+        dict_of_subset_dfs[key] = df
+
+
+# In[13]:
+
+
+# create a blank df to append the data to
+main_df = dict_of_subset_dfs["pyroptosis_shuffled_train_df"]
+# drop all rows from the df
+main_df = main_df.drop(main_df.index)
+
+
+# In[14]:
+
+
+for key in tqdm(dict_of_subset_dfs):
+    if len(dict_of_subset_dfs[key]) >= 1:
+        # loop through the dataframe
+        for cell in range(len(dict_of_subset_dfs[key])):
+            # get the first row of the dataframe
+            df = dict_of_subset_dfs[key].iloc[cell]
+            image_id = df["Metadata_ImageNumber"]
+            fov_id = df["Metadata_Site"].astype(str)
+            cell_id = df["Metadata_Cells_Number_Object_Number"]
+            well_id = df["Metadata_Well"]
+            row_id = well_id[0]
+            column_id = well_id[1:]
+            center_x = df["Metadata_Nuclei_Location_Center_X"].astype(int)
+            center_y = df["Metadata_Nuclei_Location_Center_Y"].astype(int)
+            # create a custom and contstant bounding box for the images
+            # this is made from the extracted center_x and center_y of the cell (nucleus)
+            min_x_box = center_x - radius
+            max_x_box = center_x + radius
+            min_y_box = center_y - radius
+            max_y_box = center_y + radius
+            print(cell + 1, key, row_id, column_id, fov_id, cell_id, center_x, center_y)
+
+            # create the image paths for each channel of the image
+            image_name1 = (
+                f"r{well_dict[row_id]}c{column_id}f{fov_dict[fov_id]}{image_basename_1}"
+            )
+            image_path1 = image_dir_path.joinpath(image_name1)
+
+            image_name2 = (
+                f"r{well_dict[row_id]}c{column_id}f{fov_dict[fov_id]}{image_basename_2}"
+            )
+            image_path2 = image_dir_path.joinpath(image_name2)
+
+            image_name3 = (
+                f"r{well_dict[row_id]}c{column_id}f{fov_dict[fov_id]}{image_basename_3}"
+            )
+            image_path3 = image_dir_path.joinpath(image_name3)
+
+            image_name4 = (
+                f"r{well_dict[row_id]}c{column_id}f{fov_dict[fov_id]}{image_basename_4}"
+            )
+            image_path4 = image_dir_path.joinpath(image_name4)
+
+            image_name5 = (
+                f"r{well_dict[row_id]}c{column_id}f{fov_dict[fov_id]}{image_basename_5}"
+            )
+            image_path5 = image_dir_path.joinpath(image_name5)
+
+            # crop all 5 channels of the image
+            im1 = cv2.imread(image_path1.as_posix(), cv2.IMREAD_GRAYSCALE)
+            # im_crop1 = im1[min_y_box:max_y_box, min_x_box:max_x_box]
```

Comment:
Recommend removing commented out code if not needed.

Reply:
removed